hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -384,7 +384,13 @@ object PruneDeadFields {
             f.name -> f.typ
         }: _*))
         memoizeMatrixIR(child, unify(child.typ, depMod, irDep), memo)
-      case MatrixMapRows(child, newRow, newKey) =>
+      case MatrixKeyRowsBy(child, keys, isSorted) =>
+        memoizeMatrixIR(child, child.typ.copy(
+          rvRowType = unify(child.typ.rvRowType, minimal(child.typ).rvRowType, requestedType.rvRowType),
```

Comment:
This looks right to me, but @tpoterba can you confirm?

Reply:
It can't be quite that simple, we still need to make sure the row type contains all the child key fields. I did get it slightly simpler by using `requestedType` as the base of the copy though.